### ISSUE TITLE
For func (t *JSONWebToken) Claims..., allow passing in options for the json decoder? 

### ISSUE THREAD

**Issue by aaomidi** [go-jose/go-jose#105]
For example, if I want to parse all the claims into `map[string]any` for easier looping over them, I might want the JSON decoder to use the following option: https://pkg.go.dev/encoding/json#Decoder.UseNumber

Alternatively, maybe something that can allow the user to specify what JSON decoder to use can also work. Whatever the preference is I can send in a PR for it.

**Comment by mcpherrinm** [comment]
I don't think we should make any JSON changes until at least encoding/json/v2 comes out and we have a plan to de-fork our internal JSON decoder.

JSON decoding is security-sensitive in JOSE, so I don't think it will be generally safe to accept decoder options or allow user-specified decoding.
